lib/index.d.ts

```diff
@@ -4,17 +4,21 @@
 
 /// <reference types="node" />
 
-declare function GracefulShutdown(server: any, options?: GracefulShutdown.Options): () => Promise<void>;
+declare function GracefulShutdown(
+  server: any,
+  options?: GracefulShutdown.Options
+): () => Promise<void>
 
 declare namespace GracefulShutdown {
   interface Options {
-    signals?: string;
-    timeout?: number;
-    development?: boolean;
-    forceExit?: boolean;
-    onShutdown?: (signal: string) => Promise<void>;
-    finally?: () => void;
+    signals?: string
+    timeout?: number
+    development?: boolean
+    forceExit?: boolean
+    preShutdown?: () => Promise<void>
```

Comment:
i'm hopeful that this change would land upstream, now that it is a `preShutdown` handler.

Reply:
@trane  this is a great suggestion. I would love to get your help/pair on this idea. We can access both `connectionCounter` and `connections`. In order to implement `newConnectionsAreStillBeingMade` logic in your pseudo code, we may need to keep track of the time difference between evaluation time and the last connection made timestamp. Depending on the frequency that we make readiness probe requests from k8s, we can reduce the 15s wait to something much shorter.